cloud/services/networkinterfaces/networkinterfaces_test.go

```diff
@@ -269,7 +273,9 @@ func TestReconcileNetworkInterface(t *testing.T) {
 				mPublicLoadBalancer *mock_publicloadbalancers.MockClientMockRecorder,
 				mInboundNATRules *mock_inboundnatrules.MockClientMockRecorder,
 				mInternalLoadBalancer *mock_internalloadbalancers.MockClientMockRecorder,
-				mPublicIP *mock_publicips.MockClientMockRecorder) {
+				mPublicIP *mock_publicips.MockClientMockRecorder,
+				mResourceSku *mock_resourceskus.MockClient) {
+				mResourceSku.EXPECT().HasAcceleratedNetworking(gomock.Any(), gomock.Any())
```

Comment:
Could we mock the return of HasAcceleratedNetworking on these and make sure that the CreateOrUpdate has the expected AN value as a result to unit test the logic? you can do `.Return()` to mock the return